### DISCUSSION TITLE
Treesitter in-math detect?

### DISCUSSION THREAD

**Discussion by medwatt** [L3MON4D3/LuaSnip#1023] [Q&A]
I have been growing increasingly frustrated with the default latex syntax highlight in neovim. I would rather turn it off if it wasn't for the fact that I use the function below to trigger autosnippets. The default syntax highlighting makes scrolling extremely sluggish. Treesitter highlighting has no such issue.  I was wondering if anyone knows of way to detect math zones with treesitter.

```
    function M.in_math()
        return vim.api.nvim_eval("vimtex#syntax#in_mathzone()") == 1
    end
```

**Comment by evesdropper** [comment]
pretty sure there are a few implementations that exist online already - a few I recall looking at recently at are in a [snippets plugin by iurimateus](https://github.com/iurimateus/luasnip-latex-snippets.nvim/tree/main/lua/luasnip-latex-snippets/util) and [dotfiles by kunzaatko](https://github.com/kunzaatko/nvim-dots/tree/trunk/lua/snippets/tex/utils)
